src/charm.py

```diff
@@ -315,6 +337,7 @@ def _prometheus_config(self):
 
         scrape_config = {
             "global": self._prometheus_global_config(),
+            "rule_files": ["{}/juju_*.rules".format(RULES_DIR)],
```

Comment:
Is it `.rules` or `.rule`? I hope the latter :-)

Reply:
The file gathers all the rules, provided by a single consumer into a list of alerts, this list of alerts is a "list of rules" belonging to a group, in a list of groups. The structure of this file has to conform exactly what Prometheus expects it to be - a list of `groups`.

Reply:
os.path.join

Reply:
+1, use `pathlib`

Reply:
`pathlib` done. Thanks.